sympy/core/tests/test_subs.py

```diff
@@ -640,3 +640,13 @@ def test_issue_5217():
     assert z.subs(sub) == 1 - s
     assert q == 4*x**2*y**2
     assert q.subs(sub) == 2*y**2*s
+
+
+def test_pow_eval_subs_no_cache():
+    from sympy.core.cache import clear_cache
+    from sympy.abc import x, y
+
+    s = 1/sqrt(x**2)
+    clear_cache()
+    result = s.subs(sqrt(x**2), y)
+    assert result == 1/y
```

Comment:
What does the test do if the fix in power.py is left out?
If it does not fail, the test needs a correction :-)
If it fails with an exception, we don't need the assert, we need a comment that this test is there to make sure that exception SoAndSo does not happen.
If it fails with the wrong result, the code is okay.

Also, please add the PR number in a comment (it could be inferred from the git log, but that would require more and more digging as more edits happen on the code).